bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -0,0 +1,833 @@
+package com.bugsnag.android;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * User-specified configuration storage object, contains information
+ * specified at the client level, api-key and endpoint configuration.
+ */
+@SuppressWarnings("ConstantConditions") // suppress warning about making redundant null checks
+public class Configuration implements CallbackAware, MetadataAware, UserAware {
+
+    final ConfigImpl impl;
+
+    /**
+     * Constructs a new Configuration object with default values.
+     */
+    public Configuration(@NonNull String apiKey) {
+        impl = new ConfigImpl(apiKey);
+    }
+
+    /**
+     * Loads a Configuration object from values supplied as meta-data elements in your
+     * AndroidManifest.
+     */
+    @NonNull
+    public static Configuration load(@NonNull Context context) {
+        return ConfigImpl.load(context);
+    }
+
+    @NonNull
+    static Configuration load(@NonNull Context context, @NonNull String apiKey) {
+        return ConfigImpl.load(context, apiKey);
+    }
+
+    private void warn(String property) {
+        getLogger().w("Invalid null value supplied to config." + property + ", ignoring");
+    }
+
+    /**
+     * Retrieves the API key used for events sent to Bugsnag.
+     */
+    @NonNull
+    public String getApiKey() {
+        return impl.getApiKey();
+    }
+
+    /**
+     * Changes the API key used for events sent to Bugsnag.
+     */
+    public void setApiKey(@NonNull String apiKey) {
+        if (apiKey != null) {
+            impl.setApiKey(apiKey);
+        } else {
+            warn("apiKey");
+        }
+    }
+
+    /**
+     * Sets a unique identifier for the app build to be included in all events sent to Bugsnag.
+     *
+     * This is used to identify proguard
+     * mapping files in the case that you publish multiple different apps with
+     * the same appId and versionCode. The default value is read from the
+     * com.bugsnag.android.BUILD_UUID meta-data field in your app manifest.
+     */
+    @Nullable
+    public String getBuildUuid() {
+        return impl.getBuildUuid();
+    }
+
+    /**
+     * Sets a unique identifier for the app build to be included in all events sent to Bugsnag.
+     *
+     * This is used to identify proguard
+     * mapping files in the case that you publish multiple different apps with
+     * the same appId and versionCode. The default value is read from the
+     * com.bugsnag.android.BUILD_UUID meta-data field in your app manifest.
+     */
+    public void setBuildUuid(@Nullable String buildUuid) {
+        impl.setBuildUuid(buildUuid);
```

Comment:
Could there be some merit in adding a short comment on each nullable property to say, e.g. "buildUuid is nullable".  Would just make future reviews easier and more obvious that a guard hasn't been missed off.

Reply:
Oh hang on, maybe the annotation on the parameter is sufficient - ignore that.